### PR TITLE
Check usemap attribute of <img> elements

### PR DESCRIPTION
**Pull request by igneus**
an attempt to implement #19

### CODE REVIEW

**Comment by wjdp** [comment]
@igneus This looks really good. I'll review this week.

File: htmltest/check-img.go [hunk 1]
```diff
@@ -64,6 +64,42 @@ func (hT *HTMLTest) checkImg(document *htmldoc.Document, node *html.Node) {
...
+			})
+		} else {
+			hT.checkInternalHash(usemapRef)
+		}
+
+		parent := node.Parent
+		if parent.Data == "a" {
+			hT.issueStore.AddIssue(issues.Issue{
+				Level:     issues.LevelError,
+				Message:   "img with usemap in a link",
```
Comment (wjdp) [inline htmltest/check-img.go]: Perhaps tweak the wording to explain this isn't allowed

` with usemap attribute not allowed as a descendant of am <a> tag`

File: htmltest/check-img.go [hunk 2]
```diff
@@ -64,6 +64,42 @@ func (hT *HTMLTest) checkImg(document *htmldoc.Document, node *html.Node) {
...
+				Reference: ref,
+			})
+		} else if parent.Data == "button" {
+			hT.issueStore.AddIssue(issues.Issue{
+				Level:     issues.LevelError,
+				Message:   "img with usemap in a button",
```
Comment (wjdp) [inline htmltest/check-img.go]: Same here

File: htmltest/check-img_test.go [hunk 3]
```diff
@@ -193,6 +193,42 @@ func TestImagePre(t *testing.T) {
...
+func TestImageUsemap(t *testing.T) {
+	// deals with valid usemap
+	hT := tTestFile("fixtures/images/usemapValid.html")
+	tExpectIssueCount(t, hT, 0)
+}
+
+func TestImageUsemapMapDoesNotExist(t *testing.T) {
+	// detects usemap pointing to a non-existent map
+	hT := tTestFile("fixtures/images/usemapMapDoesNotExist.html")
+	tExpectIssueCount(t, hT, 1)
```
Comment (wjdp) [inline htmltest/check-img_test.go]: Alongside `tExpectIssueCount` please also use `tExpectIssue` to ensure the correct issue is caught

(`tExpectIssue(t, hT, "message here", 1)`)

**Comment by igneus** [comment]
@wjdp All three issues mentioned in the review reflected.

File: htmltest/check-img_test.go [hunk 4]
```diff
@@ -193,6 +193,43 @@ func TestImagePre(t *testing.T) {
...
+	tExpectIssue(t, hT, "hash does not exist", 1)
+}
+
+func TestImageUsemapReferenceInvalid(t *testing.T) {
+	// detects usemap with reference formally invalid
+	hT := tTestFile("fixtures/images/usemapReferenceInvalid.html")
+	tExpectIssueCount(t, hT, 1)
```
Comment (wjdp) [inline htmltest/check-img_test.go]: Could you put in a `tExpectIssue` here?

File: htmltest/check-img_test.go [hunk 5]
```diff
@@ -193,6 +193,43 @@ func TestImagePre(t *testing.T) {
...
+}
+
+func TestImageUsemapEmpty(t *testing.T) {
+	// detects empty usemap
+	hT := tTestFile("fixtures/images/usemapEmpty.html")
+	tExpectIssueCount(t, hT, 1)
```
Comment (wjdp) [inline htmltest/check-img_test.go]: Could you put in a `tExpectIssue` here?

File: htmltest/check-img_test.go [hunk 6]
```diff
@@ -193,6 +193,43 @@ func TestImagePre(t *testing.T) {
...
+}
+
+func TestImageUsemapInLink(t *testing.T) {
+	// detects forbidden usemap in an <a> alement
+	hT := tTestFile("fixtures/images/usemapInLink.html")
+	tExpectIssueCount(t, hT, 1)
```
Comment (wjdp) [inline htmltest/check-img_test.go]: Could you put in a `tExpectIssue` here?

File: htmltest/check-img_test.go [hunk 7]
```diff
@@ -193,6 +193,43 @@ func TestImagePre(t *testing.T) {
...
+}
+
+func TestImageUsemapInButton(t *testing.T) {
+	// detects forbidden usemap in a <button> alement
+	hT := tTestFile("fixtures/images/usemapInButton.html")
+	tExpectIssueCount(t, hT, 1)
```
Comment (wjdp) [inline htmltest/check-img_test.go]: Could you put in a `tExpectIssue` here?

**Comment by igneus** [comment]
@wjdp Sure. I was surprised myself as I saw that most tests check just issue count and don't match detected issues specifically. But then I followed what seemed to be more or less standard way of writing tests in this project.

**Pull request merged**